autograd/numpy/numpy_jvps.py

```diff
@@ -1,11 +1,57 @@
+from itertools import repeat
 from . import numpy_wrapper as anp
 from .numpy_vjps import (untake, balanced_eq, match_complex, replace_zero,
                          dot_adjoint_0, dot_adjoint_1, tensordot_adjoint_0,
-                         tensordot_adjoint_1)
-from autograd.extend import defjvp, defjvp_argnum, def_linear, vspace
-from ..util import func
+                         tensordot_adjoint_1, unbroadcast_f)
+from autograd.extend import (defjvp, defjvp_argnum, def_linear, vspace,
+                             defvjp)
+from ..util import func, subval
 from .numpy_boxes import ArrayBox
 
+
+def def_ufunc_jps(ufunc, *derivs_ops):
+    derivs_ops = list(derivs_ops)
+
+    unary_ufunc_jps = {
+        'same': (lambda deriv: lambda g, ans, x:        ufunc(g),
+                 lambda deriv: lambda ans, x: ufunc),
+        'mul' : (lambda deriv: lambda g, ans, x:        g * deriv(ans, x),
+                 lambda deriv: lambda ans, x: lambda g, d=deriv(ans, x): g * d),
+        'div' : (lambda deriv: lambda g, ans, x:        g / deriv(ans, x),
+                 lambda deriv: lambda ans, x: lambda g, d=deriv(ans, x): g / d),
+        'cmul': (lambda deriv: lambda g, ans, x:        match_complex(ans, g * deriv(ans, x)),
+                 lambda deriv: lambda ans, x: lambda g, d=deriv(ans, x): match_complex(x, g * d)),
+        'cid':  (lambda deriv: lambda g, ans, x:        match_complex(ans, g),
+                 lambda deriv: lambda ans, x: lambda g: match_complex(x  , g))
+        }
+
+    if len(derivs_ops) == 1:
+        deriv, op = derivs_ops[0]
+        defjvp(ufunc, unary_ufunc_jps[op][0](deriv))
+        defvjp(ufunc, unary_ufunc_jps[op][1](deriv))
+
+    binary_ufunc_jps = {
+        'same': (lambda argnum, deriv: lambda g, ans, *args: ufunc(*subval(args, argnum, g)),
+                 lambda argnum, deriv: lambda ans, *args:
+                     unbroadcast_f(args[argnum], lambda g: ufunc(*subval(args, argnum, g)))),
+        'id':   (lambda argnum, deriv: lambda g, ans, *args: broadcast(g, ans),
+                 lambda argnum, deriv: lambda ans, *args:
+                     unbroadcast_f(args[argnum], lambda g: g)),
+        'neg':  (lambda argnum, deriv: lambda g, ans, *args: broadcast(-g, ans),
+                 lambda argnum, deriv: lambda ans, *args:
+                     unbroadcast_f(args[argnum], lambda g: -g)),
+        'mul':  (lambda argnum, deriv: lambda g, ans, *args: g * deriv(ans, *args),
+                 lambda argnum, deriv: lambda ans, *args:
+                     unbroadcast_f(args[argnum], lambda g, d=deriv(ans, *args): g * d)),
```

Comment:
For the vjps I've used this slightly weird default argument syntax to ensure that `deriv` is evaluated during the forward pass, allowing `*args` and `ans` to potentially be garbage collected.

Any objections? I could also have done this using a kind of helper closure to evaluate deriv, which would have been a bit more explicit.